### ISSUE TITLE
Installing library error: 'React/RCTBridgeModule.h' file not found

### ISSUE THREAD

**Issue by isaachinman** [ueno-llc/react-native-starter#72]
First of all, please accept my apology if this post belongs on StackOverflow or some other sort of help forum instead of here. I am new to RN so I am unaware if this problem lies in my own ignorance, or some configuration of this starter kit. If the maintainers feel this issue is not appropriate here please do close it.

I am attempting to install [react-native-share-extension](https://github.com/alinz/react-native-share-extension) into a clone of this repo. After following their setup instructions exactly, I receive the following error:

    my-app/node_modules/react-native-share-extension/ios/ReactNativeShareExtension.h:2:9: fatal error: 'React/RCTBridgeModule.h' file not found

I've spent quite some hours looking into this already. I notice you import this `RCTBridgeModule` in `ios/react-native-starter/RNUeno.h`. What is the purpose of this? Is this related, or actually unrelated?

I am aware of the [section in the docs here](https://ueno-llc.github.io/react-native-starter/#/INSTALL_DEPENDENCIES) on adding dependencies, although I'm not 100% sure it applies in this case. I've added a podspec to the above repo, making sure to detail `React` as a dependency. This has no effect on the above error.

In short, is there something special I need to do due to the structure of this starter kit?

**Comment by nova974** [comment]
I can't answer you right now. I need to test it.
If it can help you, I'm using this extension https://github.com/EstebanFuentealba/react-native-share and it's work well.
I will try to reproduce your problem this w-e. I will let you know if I find a solution.
(maybe someone else have answers ?)

**Comment by isaachinman** [comment]
@nova974 Okay, thanks! I've seen `react-native-share` as well, but it's not what I need. `react-native-share-extension` does the _opposite_, it enables data to be shared into your app from third-party apps. I specifically need image support.

Let me know if you find anything. Installing `react-native-share-extension` in a fresh `react-native init` project works fine, so it is probably something here...

**Comment by birkir** [comment]
I have used react-native-share-extension with this kit before and in general, the installaion is not a simple task.

### iOS

For iOS, you must understand that everything happens through Cocoapods (the Podfile) and can not be ignored unless you copy files from dependencies directly to your ios folder and into the XCode project.

You will need to add a new section to your Podfile with the dependencies you need.

```ruby
target 'react-native-starter' do
  ...
end

target 'react-native-starter-share' do
  pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'
  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'Core',
    'CxxBridge',
    # .. and the rest of submodules you need
  ]
end
```

The react-native-share-extension package does not have a Podspec file, so you can do two things:
a) Use the patch package method
b) Copy the files from ios folder into your project directly (ReactNativeShareExtension.h and ReactNativeShareExtension.m)

### Android

Should work out of the box

### Without react-native-share-extension for iOS

Here is a diff that could help you: https://gist.github.com/birkir/bdbbe2619b29fe0b8cf291f32868dc14

**Comment by isaachinman** [comment]
@birkir When you installed `react-native-share-extension`, any idea how you got around app containers and sharing an image url from the share extension to the main app?
